### PR TITLE
Ux vote loading

### PR DESCRIPTION
**Pull request by ggggg**
- Added a loading wheel for when the user clicks on a vote until the server responds.
- If the server doesn't respond than display error message (see image below) 

Fixes: #73

**Screenshots and screen captures:**
image
image

**Self-review checklist**

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review:

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes (what you are trying to accomplish? e.g. don't let user's age less than 0).

Completed manual review and testing of the following (please attach screenshots/gifs for UI changes and commit your test cases for functionality changes):

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

### CODE REVIEW

**Review by huonggiangbui — APPROVED** [review]
Looks good to me!

**Comment by huonggiangbui** [comment]
I also request another project owner to take a look, because I'm not sure if they want to add another library (Material UI)

**Review by embeddedt — COMMENTED** [review]
The only thing I *don't* like about this (and this might be subjective) is that it adds MUI as a dependency. My experience with MUI in another fairly large project is that it slows down TypeScript a LOT and makes autocompletion very painful to use (because of their styling system).

**Comment by ggggg** [comment]
I cannot request anyone to review, only maintainers can. It was not my idea to add MUI, @huonggiangbui requested it in #73

**Comment by huonggiangbui** [comment]
LOL you can use any library you want, MUI is just a suggestion

**Comment by huonggiangbui** [comment]
MUI is fine imo, but I want to see what Shubh or Chris think about this

**Comment by ggggg** [comment]
> LOL you can use any library you want, MUI is just a suggestion

Oh, sorry for the misunderstanding.

**Comment by huonggiangbui** [comment]
It's okay, I should make it more clear!

**Comment by hiimchrislim** [comment]
Personally, I wouldn't add in MUI. We already have Tailwind CSS and personally, I don't want to be adding in another dependency for this. For example: https://flowbite.com/docs/components/spinner/ - Just found this after a quick google search for a spinner using Tailwind

**Comment by ggggg** [comment]
> Personally, I wouldn't add in MUI. We already have Tailwind CSS and personally, I don't want to be adding in another dependency for this. For example: https://flowbite.com/docs/components/spinner/ - Just found this after a quick google search for a spinner using Tailwind

The spinner wasn't what we added MUI for, we wanted a cleaner error message. But I'll come up with something.

**Comment by ggggg** [comment]
Used alert() because this message shouldn't pop up often and to avoid adding another library/component. Please review and let me know of any changes you want me to make.

**Comment by logonoff** [comment]
If tailwind is already a dependency a better solution that might look nicer is to use one of their modal components https://tailwindui.com/components/application-ui/overlays/modals

File: client/src/pages/VotePage.tsx [hunk 1]
```diff
@@ -111,7 +125,39 @@ export const VotePage = () => {
...
+                    fill="white"
+                  />
+                  <path
+                    d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
+                    fill="#00204E"
+                  />
+                </svg>
+                <span className="sr-only">Loading...</span>
```
Comment (logonoff) [inline client/src/pages/VotePage.tsx]: do you think that the text should be more specific for screen reader users? "Loading" could mean anything

Comment (ggggg) [reply client/src/pages/VotePage.tsx]: It's literally loading, they should know given the context of their more recent action. As discussed in #57 we might want to use this loading on page refresh.

**Comment by ggggg** [comment]
> If tailwind is already a dependency a better solution might look nicer to use one of their modal components https://tailwindui.com/components/application-ui/overlays/modals

I will look into it, thanks!

**Comment by ggggg** [comment]
@logonoff Sorry for this taking a while, I had midterms and was busy with other school stuff.
Anyways, this should now use tailwind's alert box instead of the ugly javascript one.
Here is a gif preview:
gif_preview

**Comment by embeddedt** [comment]
I know this is incredibly picky, but I'm not liking that the spinner isn't centered around the C. :laughing: Is it easy to fix that?

The alert box looks great!

**Comment by ggggg** [comment]
Well the spinner is centered, I think anything else would look odd. A possible fix can be to mess around with the size of the spinner.

**Comment by ggggg** [comment]
Are any of these any better @embeddedt?
chrome_spNEvwLG2X
chrome_X0FsL5nITg

**Comment by embeddedt** [comment]
The issue is less obvious now but it's still not centered compared to the center of the C. I'm not sure why. It's weird, since I assume the buttons have centered text and the spinner is also centered.

File: client/src/pages/VotePage.tsx [hunk 2]
```diff
@@ -111,7 +127,47 @@ export const VotePage = () => {
...
+            <div
+              className={
+                "flex items-center w-full bg- h-full flex-col justify-center align-middle"
+              }
+              style={{ backgroundColor: "rgba(0,0,0,0.5)" }}
+            >
+              <div role="status">
+                <svg
+                  aria-hidden="true"
+                  className="mr-2 w-8 h-8 text-gray-200 animate-spin dark:text-gray-900 fill-blue-600"
```
Comment (embeddedt) [inline client/src/pages/VotePage.tsx]: This fixes it for me.

```suggestion
                  className="w-8 h-8 text-gray-200 animate-spin dark:text-gray-900 fill-blue-600"
```

**Comment by embeddedt** [comment]
Also, I like the first design in https://github.com/utmgdsc/PollVotingSystem/pull/74#issuecomment-1288165720 the best (the one with no background), as it looks cleaner IMO and should also be easier to port to a future dark mode.

**Comment by ggggg** [comment]
Hey is this still open?

**Review by logonoff — APPROVED** [review]
looks fine to me

**Review by shubhbapna — APPROVED** [review]
LGTM

**Pull request merged**